usr/iscsid.c

```diff
@@ -590,10 +613,12 @@ int main(int argc, char *argv[])
 
 		pid = fork();
 		if (pid == 0) {
+			LIST_HEAD(sync_list);
 			int nr_found; /* not used */
 			/* child */
-			/* TODO - test with async support enabled */
-			iscsi_sysfs_for_each_session(NULL, &nr_found, sync_session, 0);
+			iscsi_sysfs_for_each_session(&sync_list, &nr_found,
+						     init_sync_session_req, 0);
+			sync_sessions(&sync_list);
```

Comment:
I suppose you can optimize this by adding "if (nr_found)" before the call to sync_sessions(), correct? Since, usually, there will be none to sync. Not sure I feel strongly about this, though. :)

Reply:
I'll the check. We could have iscsi_sysfs_count_sessions return > 0, but then init_sync_session_req didn't set anything up.